flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartitionReadSchedulerTest.java

```diff
@@ -192,6 +199,131 @@ public void testOnReadBufferRequestError() throws Exception {
+                new FakedSortMergeResultPartitionReadScheduler(
+                        bufferPool, executor, this, bufferRequestTimeout);
+
+        SortMergeSubpartitionReader subpartitionReader0 =
+                readScheduler.crateSubpartitionReader(
+                        new NoOpBufferAvailablityListener(), 0, partitionedFile);
+        SortMergeSubpartitionReader subpartitionReader1 =
+                readScheduler.crateSubpartitionReader(
+                        new NoOpBufferAvailablityListener(), 1, partitionedFile);
+
+        PriorityQueue<SortMergeSubpartitionReader> allReaders = new PriorityQueue<>();
+        allReaders.add(subpartitionReader0);
+        allReaders.add(subpartitionReader1);
+
+        long startTimestamp = System.nanoTime();
+        Queue<MemorySegment> allocatedBuffers = readScheduler.allocateBuffers(allReaders);
+        long requestDuration = System.nanoTime() - startTimestamp;
+
+        assertEquals(2, allocatedBuffers.size());
+        assertTrue(requestDuration > bufferRequestTimeout.toNanos() * 2);
+        assertNull(subpartitionReader0.getFailureCause());
+        assertNull(subpartitionReader1.getFailureCause());
+
+        bufferPool.destroy();
+        readScheduler.release();
+    }
+
+    private static void assertExpectedTimeoutException(Throwable throwable) {
+        assertNotNull(throwable);
+        assertTrue(
+                ExceptionUtils.findThrowableWithMessage(throwable, "Buffer request timeout")
+                        .isPresent());
+    }
+
+    private static class FakedBatchShuffleReadBufferPool extends BatchShuffleReadBufferPool {
+        private List<MemorySegment> requestedBuffers;
+        private final boolean isRecycleOnlyOneBuffer;
+        private boolean isFirstlyCallGetTimestamp;
+        private int recycleBufferIndex;
+
+        FakedBatchShuffleReadBufferPool(
+                long totalBytes, int bufferSize, boolean isRecycleOnlyOneBuffer) throws Exception {
+            super(totalBytes, bufferSize);
+
+            this.isRecycleOnlyOneBuffer = isRecycleOnlyOneBuffer;
+            this.requestedBuffers = requestBuffers();
+            this.isFirstlyCallGetTimestamp = true;
+            this.recycleBufferIndex = 0;
+        }
+
+        @Override
+        public long getLastBufferOperationTimestamp() {
+            if (isFirstlyCallGetTimestamp) {
+                isFirstlyCallGetTimestamp = false;
+                return super.getLastBufferOperationTimestamp();
+            }
+
+            if (recycleBufferIndex == 0 || (recycleBufferIndex == 1 && !isRecycleOnlyOneBuffer)) {
+                recycle(requestedBuffers.get(recycleBufferIndex++));
+            }
+            return super.getLastBufferOperationTimestamp();
+        }
+
+        @Override
+        public void destroy() {
+            requestedBuffers.clear();
+            super.destroy();
+        }
+    }
+
+    private static class FakedSortMergeResultPartitionReadScheduler
```

Comment:
I guess we can remove this FakedSortMergeResultPartitionReadScheduler and create the SortMergeSubpartitionReader instance by calling the constructor directly.

Reply:
Yes, you're right. Removed it.